lib/src/cpp/utils/XMLManager.hpp

```diff
@@ -151,14 +197,14 @@ class XMLManager
      * @throw ElementNotFound exception if expected node was not found and node creation was not required
      * @throw BadParameter exception if expected node could not be found by using the given index.
      */
-    void get_node(
+    void move_to_node(
             const std::string& index,
             const std::string& default_tag_name,
             const bool create_if_not_existent);
 
     /**
-     * @brief Get the node object that matches the given tag name, and has the same attribute key-value pair
-     *        set as the given name-value pair.
+     * @brief Move the reference node to the node object that matches the given tag name, and has the same attribute
+     *        key-value pair set as the given name-value pair.
```

Comment:
I suppose that similar to the index node, if and empty key is passed, we should keep in the same node. We will refector it but not in this PR. Just so we can keep track. Leave a TODO note, please.